tests/core/anthropic/test_stream.py

```diff
@@ -122,17 +124,228 @@ def generator():
+            index=1,
+            type="content_block_delta",
+        ),
+        RawContentBlockStopEvent(index=1, type="content_block_stop"),
+        RawMessageDeltaEvent(
+            delta=Delta(stop_reason="tool_use", stop_sequence=None),
+            type="message_delta",
+            usage=MessageDeltaUsage(output_tokens=1),
+        ),
+        RawMessageStopEvent(type="message_stop"),
+    ]
+
+    def generator():
+        for chunk in chunks:
+            call_response_chunk = AnthropicCallResponseChunk(chunk=chunk)
+            if (
+                isinstance(call_response_chunk.chunk, RawContentBlockStartEvent)
+                and (tool_call := call_response_chunk.chunk.content_block)
+                and tool_call.type == "tool_use"
+            ):
+                tool_call = ToolUseBlock(
+                    id="tool_id",
+                    input={
+                        "title": "Sapiens: A Brief History of Humankind",
+                        "author": "Harari, Yuval Noah",
+                    },
+                    name="FormatBook",
+                    type="tool_use",
+                )
+                yield (
+                    call_response_chunk,
+                    FormatBook.from_tool_call(tool_call),
+                )
+            else:
+                yield call_response_chunk, None
+
+    stream = AnthropicStream(
+        stream=generator(),
+        metadata={},
+        tool_types=[FormatBook],
+        call_response_type=AnthropicCallResponse,
+        model="claude-3-5-sonnet-20240620",
+        prompt_template="",
+        fn_args={},
+        dynamic_config=None,
+        messages=[{"role": "user", "content": "content"}],
+        call_params=AnthropicCallParams(max_tokens=1000),
+        call_kwargs={},
+    )
+
+    for _ in stream:
+        pass
+
+    tool_call = ToolUseBlock(
+        id="tool_id",
+        input={
+            "title": "Sapiens: A Brief History of Humankind",
+            "author": "Harari, Yuval Noah",
+        },
+        name="FormatBook",
+        type="tool_use",
+    )
+    content = TextBlock(text="content", type="text")
+    completion = Message(
+        id="id",
+        content=[content, tool_call],
+        model="claude-3-5-sonnet-20240620",
+        role="assistant",
+        stop_reason="end_turn",
+        stop_sequence=None,
+        type="message",
+        usage=Usage(input_tokens=1, output_tokens=2),
+    )
+    call_response = AnthropicCallResponse(
+        metadata={},
+        response=completion,
+        tool_types=[FormatBook],
+        prompt_template="",
+        fn_args={},
+        dynamic_config=None,
+        messages=[],
+        call_params=AnthropicCallParams(max_tokens=1000),
+        call_kwargs={},
+        user_message_param=None,
+        start_time=0,
+        end_time=0,
+    )
+    constructed_call_response = stream.construct_call_response()
+
+    assert constructed_call_response._provider == call_response._provider
```

Comment:
why not just assert the model dumps are equal?

Reply:
i dont copy over everything to stream when constructing the call response, we would need to implement provider specific properties to save all of them when we iterate chunks when we construct call params

Reply:
done